blivet/fstab.py

```diff
@@ -0,0 +1,792 @@
+            :keyword vfstype: device file system type (see man fstab(5) fs_vfstype)
+            :type vfstype: str
+            :keyword mntops: list of mount option strings (see man fstab(5) fs_mntops)
+            :type mnops: list
+            :keyword freq: whether to dump the filesystem (see man fstab(5) fs_freq)
+            :type freq: int
+            :keyword passno: fsck order or disable fsck if 0 (see man fstab(5) fs_passno)
+            :type passno: int
+            :keyword comment: multiline comment added to fstab before entry; each line will be prefixed with "# "
+            :type comment: str
+            :keyword entry: fstab entry as an alternative input type
+            :type: :class: `FSTabEntry`
+        """
+
+        # Default mount options
+        if mntops is None:
+            mntops = ['defaults']
+
+        # Use existing FSTabEntry or create a new one
+        _entry = entry or FSTabEntry()
+
+        if spec is not None:
+            _entry.spec = spec
+
+        if file is not None:
+            _entry.file = file
+
+        if vfstype is not None:
+            _entry.vfstype = vfstype
+
+        if mntops is not None:
+            _entry.mntops = mntops
+        elif _entry.mntops is None:
+            _entry.mntops = ['defaults']
+
+        if freq is not None:
+            # Whether the fs should be dumped by dump(8) (default: 0, i.e. no)
+            _entry.freq = freq
+        elif _entry.freq is None:
+            _entry.freq = 0
+
+        if passno is not None:
+            _entry.passno = passno
+        elif _entry.passno is None:
+            # 'passno' represents order of fsck run at the boot time (default: 0, i.e. disabled).
+            # '/' should have 1, other checked should have 2
+            if _entry.file == '/':
+                _entry.passno = 1
+            elif _entry.file.startswith('/boot'):
+                _entry.passno = 2
+            else:
+                _entry.passno = 0
+
+        if comment is not None:
+            # Add '# ' at the start of any comment line and newline to the end of comment.
+            # Has to be done here since libmount won't do it.
+            modif_comment = '# ' + comment.replace('\n', '\n# ') + '\n'
+            _entry.comment = modif_comment
+
+        self._table.add_fs(_entry.entry)
+
+    def remove_entry(self, spec=None, file=None, *, entry=None):
+        """ Find and remove entry from fstab based on spec/file.
+            If both entry and spec/file are given, spec/file are prioritized over entry values.
+
+            :keyword spec: device specs (see man fstab(5) fs_spec)
+            :type spec: str
+            :keyword file: device mount path (see man fstab(5) fs_file)
+            :type file: str
+            :keyword entry: fstab entry as an alternative input type
+            :type: :class: `FSTabEntry`
+        """
+
+        fs = self.find_entry(spec, file, entry=entry)
+        if fs:
+            self._table.remove_fs(fs.entry)
```

Comment:
This should raise an exception when the entry is not found.

Reply:
Fixed. Raises ValueError if no matching entry is found in fstab.